lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -2584,13 +2585,27 @@ struct TritonGPUInferLayoutInterface
         return failure();
       }
       resultEncoding = BlockedEncodingAttr::get(
-          getDialect()->getContext(),
-          applyPermutation(enc.getSizePerThread(), order),
+          ctx, applyPermutation(enc.getSizePerThread(), order),
           applyPermutation(enc.getThreadsPerWarp(), order),
           applyPermutation(enc.getWarpsPerCTA(), order),
           applyPermutation(invOrderUnsigned, enc.getOrder()), *ctaLayout);
       return success();
     }
+    if (auto ll = toLinearLayout(shape, operandEncoding)) {
+      auto permutedDims =
+          applyPermutation(llvm::to_vector(ll->getOutDimNames()), order);
+      // transposedLl has transposed dims
+      auto transposedLl = ll->transposeOuts(permutedDims);
+      auto rank = shape.size();
+      SmallVector<std::pair<StringAttr, int32_t>> reorderedDims;
+      for (auto [dimi, permDim] :
+           llvm::zip(standardOutDimNames(ctx, rank), permutedDims)) {
+        reorderedDims.emplace_back(dimi, transposedLl.getOutDimSize(permDim));
+      }
+      auto retLl = transposedLl.reshapeOuts(reorderedDims);
+      resultEncoding = LinearEncodingAttr::get(ctx, retLl);
+      return success();
+    }
 
     return failure(); // unhandled encoding
```

Comment:
Is it possible to remove the failure case now?

Reply:
There is still one layout that I know of (`DotOpererand(Blocked)`) that does not have a a to LL function. Once we support this one (it's my next task after I land these PRs pretty much) we'll start the clean-up and DCE of all these paths.